virtual_rainforest/core/constants_class.py

```diff
@@ -0,0 +1,38 @@
+"""A base constants dataclass."""
+
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass, fields
+from typing import Any
+
+from virtual_rainforest.core.exceptions import ConfigurationError
+from virtual_rainforest.core.logger import LOGGER
+
+
+@dataclass(frozen=True)
+class ConstantsDataclass(ABC):
+    """XYZ."""
+
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> ConstantsDataclass:
+        """MNO."""
```

Comment:
These need proper docstrings

Reply:
Oh god yes, but all of the docstrings and docs need rewriting too. Once the implementation has settled down, I'll do that.

Reply:
It does also help with reviewing if you can provide a docstring, even if it's just one line. Adding dummy docstrings like this just seems like an exercise in making `flake8` shut up